IoTCIntegration/lib/engine.js

```diff
@@ -75,14 +75,29 @@ function validateMeasurements(measurements) {
     }
 
     for (const field in measurements) {
-        if (typeof measurements[field] !== 'number' && typeof measurements[field] !== 'string') {
+        if (typeof measurements[field] !== 'number' && typeof measurements[field] !== 'string' && !isLocation(measurements[field])) {
             return false;
         }
     }
 
     return true;
 }
 
+/**
+ * @returns true if a measurement is a location.
+ */
+function isLocation(measurement) {
+    if (typeof measurement !== 'object' || typeof measurement.lat !== 'number' || typeof measurement.lon !== 'number') {
+        return false;
+    }
+
+    if ('alt' in measurement && typeof measurement.alt !== 'number') {
```

Comment:
not sure how strict we want the validation of locations to be, but do we care if they have extra fields in the object other than `lat`, `lon` and `alt`?

Reply:
We don't care if there are extra fields. It's fine if users have a more complex location object in their side.